### PR TITLE
Point `node_modules/bs-platform` to `melange` to disguise `melange` as `bsb`

### PR DESCRIPTION
**Pull request by bhootd**
Would you be willing to merge this change? I would rather have this in the original template instead of maintaining a fork.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,6 @@ _esy
 .bsb.lock
 .merlin
 dune
-dune.bsb
\ No newline at end of file
+dune.bsb
+esy.lock
```
Comment (jchavarri) [inline .gitignore]: I don't think we want to ignore `esy.lock` folder, checking this folder in git makes sure everyone has the same versions of everything.

File: README.md [hunk 2]
```diff
@@ -20,10 +20,15 @@ Then:
 esy build
 ```
 
-to build the project. Now you should see a `_build` folder with all generated files, you can run
+to build the project. This command, as constructed in `esy.json`,
+
+1. generates a symbolically linked dir `bs-platform` in `node_modules`, which effectively makes `melange` the compiler behind the old `bsb` command, and then
```
Comment (jchavarri) [inline README.md]:
```suggestion
1. generates a symbolically linked dir `bs-platform` in `node_modules`, which is helpful when using JavaScript bundlers like Webpack that expect to find Melange stdlib JavaScript files under `node_modules/bs-platform`.
```

File: README.md [hunk 3]
```diff
@@ -20,10 +20,15 @@ Then:
...
+2. builds the project using the symbolically linked `bsb` command.
```
Comment (jchavarri) [inline README.md]:
```suggestion
2. builds the project.
```

**Review by jchavarri — CHANGES_REQUESTED** [review]
Thanks for this contribution! ✨ 

Please take a look at the requested changes, let me know what you think :)

File: .gitignore (hunk 1 above)
Comment (bhootd) [reply .gitignore]: Wasn't aware of this. I have pushed a commit to this effect.

**Review by jchavarri — APPROVED** [review]
Thanks!

**Pull request merged**